src/Radicle/Daemon/Common.hs

```diff
@@ -0,0 +1,66 @@
+module Radicle.Daemon.Common
+  ( ReaderOrWriter(..)
+  , Polling(..)
+  , Chain(..)
+  , Chains(..)
+  , logInfo
+  , logErr
+  , advanceChain
+  ) where
+
+import           Protolude hiding (log)
+
+import qualified Data.Aeson as A
+import qualified Data.Sequence as Seq
+import qualified Data.Text as T
+import qualified Data.Time.Clock.System as Time
+
+import           Radicle
+import qualified Radicle.Internal.ConcurrentMap as CMap
+
+data ReaderOrWriter = Reader | Writer
+  deriving (Generic)
+
+data Polling = HighFreq Int64 | LowFreq
+
+instance A.ToJSON ReaderOrWriter
+instance A.FromJSON ReaderOrWriter
+
+data Chain id idx subs = Chain
```

Comment:
Can we replace the term “Chain” with “Machine”

Reply:
What is the reason behind parameterizing the type?

Reply:
It was to make it useable in both the server and the daemon. The shared code turned out to be very small though; not sure it's worth it.

Reply:
Let’s not share code between the server and the daemon. It is very little code and we are going to kill the server. So copy paste is fine.